src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java

```diff
@@ -143,9 +145,17 @@ private boolean shouldSab() {
     }
 
     private boolean shouldRsb() {
-        if (!main.config.LOOT.RSB.ENABLED || !target.npcInfo.extra.has(NpcExtra.USE_RSB)) return false;
-        if (usedRsb < System.currentTimeMillis() - main.config.LOOT.RSB.AMMO_REFRESH) usedRsb = System.currentTimeMillis();
-
+        if (!main.config.LOOT.RSB.ENABLED || main.config.LOOT.RSB.KEY == null
+                || !target.npcInfo.extra.has(NpcExtra.USE_RSB)) return false;
+        String itemId = main.facadeManager.slotBars.standardBar.slots.stream()
+                .filter(slot -> Objects.equals(main.facadeManager.settings.getCharCode(valueOf
+                        ("SLOTBAR_" + (slot.slotNumber == 10 ? 0 : slot.slotNumber))), main.config.LOOT.RSB.KEY))
+                .findFirst().map(slot -> slot.item != null ? slot.item.id : null).orElse(null);
+        if (!Objects.equals(itemId, "ammunition_laser_rsb-75")
+                && !Objects.equals(itemId, "ammunition_laser_rcb-140")) return false; // TODO: Check bool cooldown from enum, after API gets merged.
+        boolean isReady = bar.findItemById(itemId).map(i -> i.activatable).orElse(false);
```

Comment:
Add this method to SettingsProxy:
```java
@Nullable
public KeyBind getKeyBind(Character ch) {
    if (ch == null) return null;
    for (int i = 0; i < keycodes.length; i++) {
        if (keycodes[i] == ch) return KeyBind.values()[i];
    }
    return null;
}
```

Add this method to SlotBarsProxy (as well as uncommenting premiumBar):
```java
@Nullable
public Slot getSlot(KeyBind key) {
        if (key == null) return null;
        String keyStr = keybind.toString();
        
        SlotBar sb = keyStr.startsWith("SLOTBAR") ? standardBar :
                keyStr.startsWith("PREMIUM") ? premiumBar : null;
        
        return sb == null ? null : sb.slots.get(Integer.parseInt(keyStr.split("_")[1]));
}
```

Then this code can be simplified to just:
```suggestion
        KeyBind key = settings.getKeyBind(main.config.LOOT.RSB.KEY);
        Slot slot = slotBars.get(key);
        if (slot == null || slot.item == null) return false;
        boolean isReady = bar.findItemById(slot.item.id).map(i -> i.activatable).orElse(false);
```

Reply:
Will do, after you decide that you want this approach or https://github.com/ODAKAB/DarkBot/commit/da4319845ac670015feb93a51811d5e02b731f2f fixed.

Reply:
I think it's best for it to be toggling on & off the ammo if you happen to pick an RSB ammo that isn't rsb, and it to work fine if you happen to put another random ability in there

Reply:
I mean
```
public String getItemIdByKey(Character c) {
        SettingsProxy.KeyBind keybind = main.facadeManager.settings.getAtChar(c);
        if (keybind == null) return null;

        if (keybind.ordinal() <= 9) {
            return standardBar.slots.stream().filter(s -> s.slotNumber == keybind.ordinal() + 1)
                    .filter(s -> s.item != null).map(s -> s.item.id).findFirst().orElse(null);
        }
        else return null;
    }
```
in SlotbarsProxy of old commit

Reply:
simply use the methods i provided, this one doesn't handle premium bar

Reply:
Okay I'll start applying today. But this was going to handle premium bar and anything in future just with keybind.ordinal value.